### PR TITLE
Add cppzmq 4.7.1

### PR DESCRIPTION
**Pull request by gummif**
Specify library name and version:  **cppzmq/4.7.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cppzmq/all/patches/0002-cmakelists-alias-libzmq-no-export.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
++++ b/CMakeLists.txt
+@@ -8,6 +8,11 @@ project(cppzmq VERSION ${DETECTED_CPPZMQ_VERSION})
+ 
+ if (NOT TARGET libzmq AND NOT TARGET libzmq-static)
+   find_package(ZeroMQ QUIET)
++  add_library(libzmq INTERFACE)
++  target_link_libraries(libzmq INTERFACE ZeroMQ::ZeroMQ)
++  add_library(libzmq-static INTERFACE)
++  target_link_libraries(libzmq-static INTERFACE ZeroMQ::ZeroMQ)
++  install(TARGETS libzmq libzmq-static EXPORT ${PROJECT_NAME}-targets)
```
[the remark this answers is not in the document]
Comment (madebr) [reply recipes/cppzmq/all/patches/0002-cmakelists-alias-libzmq-no-export.patch:11]: Indeed, the patch is not needed.

[the remark this answers is not in the document]
Comment (gummif) [reply recipes/cppzmq/all/patches/0002-cmakelists-alias-libzmq-no-export.patch]: Is this not needed at all, or just the intall statement? My understanding is that libzmq is exported as ZeroMQ::ZeroMQ by conan and therefore this is needed?

[the remark this answers is not in the document]
Comment (madebr) [reply recipes/cppzmq/all/patches/0002-cmakelists-alias-libzmq-no-export.patch]: I checked again and I was wrong.

**Review by madebr — DISMISSED** [review]

File: recipes/cppzmq/all/patches/0002-cmakelists-alias-libzmq-no-export.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (gummif) [reply recipes/cppzmq/all/patches/0002-cmakelists-alias-libzmq-no-export.patch]: Ok, looks like this line is needed after all for some reason. I should submit a PR to cppzmq to make this less error prone in the future.

**Comment by gummif** [comment]
`install(...)` in the patch is required, otherwise the build errors on
```
CMake Error: install(EXPORT "cppzmq-targets" ...) includes target "cppzmq" which requires target "libzmq" that is not in any export set.
CMake Error: install(EXPORT "cppzmq-targets" ...) includes target "cppzmq-static" which requires target "libzmq-static" that is not in any export set.
CMake Error in source_subfolder/CMakeLists.txt:
  export called with target "cppzmq" which requires target "libzmq" that is
  not in any export set.


CMake Error in source_subfolder/CMakeLists.txt:
  export called with target "cppzmq-static" which requires target
  "libzmq-static" that is not in any export set.
```

File: recipes/cppzmq/all/patches/0002-cmakelists-alias-libzmq-no-export.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/cppzmq/all/patches/0002-cmakelists-alias-libzmq-no-export.patch:11]: Okay, keep the export

**Review by madebr — APPROVED** [review]

**Review by gocarlos — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**